middleware/middleware.go

```diff
@@ -0,0 +1,145 @@
+package middleware
+
+import (
+	//User-defined packages
+
+	"online/helper"
+	"online/logs"
+	"online/models"
+	"online/repository"
+
+	//Inbuild packages
+	"errors"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	//Third-party packages
+	"github.com/dgrijalva/jwt-go"
+	"github.com/labstack/echo"
+	"gorm.io/gorm"
+)
+
+// Create a JWT token with the needed claims
+func CreateToken(user models.User, c echo.Context) (string, error) {
+	log := logs.Log()
+	if err := helper.Config(".env"); err != nil {
+		log.Error.Println("Error : 'Error at loading '.env' file'")
+	}
+	exp := time.Now().Add(time.Hour * 24).Unix()
+	userId := strconv.Itoa(int(user.UserId))
+	roleId := strconv.Itoa(int(user.RoleId))
+	claims := jwt.MapClaims{
+		"ExpiresAt": exp,
+		"User-id":   userId,
+		"IssuedAt":  time.Now().Unix(),
+		"Role-id":   roleId,
+	}
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	tokenString, err := token.SignedString([]byte(os.Getenv("SECRET_KEY")))
+	if err != nil {
+		return "", err
+	}
+	return tokenString, nil
+}
+
+type Database struct {
+	Db *gorm.DB
+}
+
+// Token and claims validation
+func (db Database) AuthMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
+	log := logs.Log()
+	if err := helper.Config(".env"); err != nil {
+		log.Error.Println("Error : 'Error at loading '.env' file'")
+	}
+	return func(c echo.Context) error {
+		tokenString := c.Request().Header.Get("Authorization")
+		//To check the token is empty or not
+		if tokenString == "" {
+			return c.JSON(http.StatusBadRequest, map[string]interface{}{
+				"Status": 400,
+				"Error":  "token is empty",
+			})
+		}
+
+		for index, char := range tokenString {
+			if char == ' ' {
+				tokenString = tokenString[index+1:]
+			}
+		}
+
+		claims := jwt.MapClaims{}
+		token, err := jwt.ParseWithClaims(tokenString, &claims, func(token *jwt.Token) (interface{}, error) {
+			return []byte(os.Getenv("SECRET_KEY")), nil
+		})
+
+		if err != nil {
+			if errors.Is(err, jwt.ErrSignatureInvalid) {
+				return c.JSON(http.StatusBadRequest, map[string]interface{}{
+					"Status": 400,
+					"Error":  "Invalid token signature",
+				})
+			} else if claims["ExpiresAt"].(int64) < time.Now().Unix() {
+				repository.DeleteToken(db.Db, claims["User-id"].(string))
+				log.Error.Println("Error : 'session expired...login again!!!' Status : 440")
```

Comment:
What is the purpose of "440" status code, Here you need to use "401 Unauthorized"

Reply:
"440" status code indicates session expired error. So that I've used here.